### PR TITLE
GH-49: Add MessageConversion and JSON

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves: #49

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -143,12 +144,9 @@ project ('spring-kafka') {
 
 	dependencies {
 		compile "org.springframework:spring-messaging:$springVersion"
-//		compile ("org.apache.avro:avro:$avroVersion", optional)
-//		compile ("org.apache.avro:avro-compiler:$avroVersion", optional)
-//		compile "com.goldmansachs:gs-collections:$gsCollectionsVersion"
-//		compile "io.projectreactor:reactor-core:$reactor2Version"
```
Comment (artembilan) [inline build.gradle]: If we remove the deps, we probably should remove their versions as well.

File: spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessageConverter.java [hunk 2]
```diff
@@ -16,21 +16,36 @@
...
  *
  * @author Gary Russell
  */
-public interface MessageConverter<K, V> {
+public interface MessageConverter {
+
+	/**
+	 * Convert a {@link ConsumerRecord} to a {@link Message}.
+	 * @param record the record.
+	 * @param acknowledgment the acknowledment.
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessageConverter.java]: Typo: acknowled**g**ment

File: spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessagingMessageConverter.java [hunk 3]
```diff
@@ -37,7 +39,7 @@
  * @author Marius Bogoevici
  * @author Gary Russell
  */
-public class MessagingMessageConverter<K, V> implements MessageConverter<K, V> {
+public class MessagingMessageConverter<K, V> implements MessageConverter {
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessagingMessageConverter.java]: My IDEA says me that the generics are redundant.
Investigating why you need them at all...

File: spring-kafka/src/main/java/org/springframework/kafka/support/converter/StringJsonMessageConverter.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+/**
+ * JSON Message converter - String on output, String or byte[] on input.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Gary Russell
+ *
+ */
+public class StringJsonMessageConverter<K, V> extends MessagingMessageConverter<K, V> {
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/support/converter/StringJsonMessageConverter.java]: Well, looks like we are fine even without all these generics.
Am I missing anything ?

Comment (garyrussell) [inline spring-kafka/src/main/java/org/springframework/kafka/support/converter/StringJsonMessageConverter.java]: As I mentioned last night I was struggling with Generics yesterday; when I propagated `MessageConverter` through a bunch of APIs I was forced to add generics to lots of classes so I ended up removing them; I am not surprised there are some leftovers.

I'll take a look.

**Comment by garyrussell** [comment]
Pushed

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/a087ce21b3e28eb3b199e0b01e4e8f246a62b4ae after rebase.

**Pull request closed without merging**